sandbox/lua/encoders/schema_influx.lua

```diff
@@ -176,13 +170,85 @@ function process_message()
         series = string.gsub(series_orig, "%%{([%w%p]-)}", sub_func)
     end
 
-    local output = {
+    output = {
         {
             name = series,
             columns = columns,
             points =  {values}
         }
     }
+
+    return output
+end
+
+local function process_multi_series(output, columns, values, place)
+    local output_index = 1
+    while true do
+        -- Iterate through Fields array in the message
+        local typ, name, value, representation, count = read_next_field()
+        -- Exit the perpetual loop when the iteration is complete
+        if not typ then break end
+
+        -- Only process fields that are not requested to be skipped
+        if not skip_fields_str or not skip_fields[name] then
+            local field_name = ""
+            local field_columns = {}
+            local field_points = {}
+
+            -- Instantiate a table in the output table for this iteration
+            output[output_index] = {}
+
+            -- Set the name attribute of this table appropriately
+            if use_subs then
+                series = string.gsub(series_orig, "%%{([%w%p]-)}", sub_func).."."..name
+            end
+            field_name = series
+
+            -- Create new tables built from local columns, values in this table
+            for k,v in pairs(columns) do field_columns[k] = v end
+            for k,v in pairs(values) do field_points[k] = v end
+
+            -- Merge added values to each table for the field in this iteration
+            field_columns[place] = "value"
+            field_points[place] = value
+            -- Structure the table to match the expected Influxdb structure
+            output[output_index] = {
+                name = field_name,
+                columns = field_columns,
+                points = {field_points}
+            }
+            -- Increment the index to the table so the next iteration will
+            -- append another entry to the array. This allows us to send a
+            -- bunch of metrics to InfluxDB with a single HTTP request
+            output_index = output_index + 1
+        end
+    end
+
+    return output
+end
+
+function process_message()
+    local output = {}
+    local columns = {}
+    local values = {}
+
+    columns[1] = "time" -- InfluxDB's default
+    values[1] = read_message("Timestamp") / 1e6
+
+    local place = 2
+    for _, field in ipairs(used_base_fields) do
+        columns[place] = field
+        values[place] = read_message(field)
+        place = place + 1
+    end
+
+    local multi_series = read_config("multi_series")
```

Comment:
This should be called outside of `process_message` with the other `read_config` call so it doesn't have to be done for every message.